File: backend/app/Models/AccountType.php
<?php

namespace App\Models;

use Exception;
use Illuminate\Database\Eloquent\Model;

class AccountType extends Model
{
    protected $fillable = [
        'account_group_id',
        'account_type_name',
    ];

    public function accountGroup()
    {
        return $this->belongsTo(AccountGroup::class, 'account_group_id');
    }

    public function accounts()
    {
        return $this->hasMany(Account::class, 'account_type_id');
    }
}

File: backend-optimized/models/AccountType.go
package models

type AccountType struct {
	ID              uint         `gorm:"primaryKey"`
	AccountGroupID  uint         `gorm:"column:account_group_id"`
	AccountGroup    AccountGroup `gorm:"foreignKey:AccountGroupID"`
	AccountTypeName string       `gorm:"column:account_type_name"`
}
